server/jetstream_api.go

```diff
@@ -667,6 +675,19 @@ type JSApiConsumerDeleteResponse struct {
 
 const JSApiConsumerDeleteResponseType = "io.nats.jetstream.api.v1.consumer_delete_response"
 
+type JSApiConsumerPauseRequest struct {
+	PauseUntil time.Time `json:"pause_until,omitempty"`
```

Comment:
@neilalexander should we make this a pointer also? Ditto in the response?

Reply:
In the config it makes sense because it means we just don't store it in the config JSON if it's a truly empty pause, but I'm not sure if it does here. I'd be interested in your thoughts.

The only input to the pause request is the timestamp so it's hardly optional and will already have a zero default. Likewise the response should ideally be an echo of whatever the new state is, paused or unpaused.

Reply:
There's one scenario I can think of where returning the timestamp always makes sense and that's dealing with timezones. The server does everything here in UTC, so if I sent a timestamp with a TZ offset that I thought would cause a pause but didn't, I'd want to know what the server converted it to. That might help to identify a configuration problem somewhere.

Reply:
Yeah good point, thanks @neilalexander